test-suite/src/test/scala/org/scalajs/testsuite/compiler/RegressionTest.scala

```diff
@@ -444,4 +450,22 @@ object RegressionTest extends JasmineTest {
       expect(f(5)).toEqual(6)
     }
   }
+
+  class Bug1955 {
+    var result: Int = 0
+
+    def doSomething[A](a: Int, b: Int, r: A): A = {
+      result = a + b
+      r
+    }
+
+    def bug(e: Boolean): Unit = {
+      1 match {
+        case 1 => doSomething(123, 456, ())
+        case 2 if e =>
```

Comment:
We should have tests that execute code that is in the branch with the fixed `if`/`block`.
```scala 
bug.bug(1, true)
expect(bug.result).toEqual(579)
bug.bug(2, true) // Does nothing
expect(bug.result).toEqual(579)
bug.bug(2, false) // should throw the match error.
...
def bug(x:Int, e: Boolean): Unit = {
  x match {
    case 1 => doSomething(123, 456, ())
    case 2 if e => 
  }
}
```